src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -647,4 +648,15 @@ inline ShenandoahMarkingContext* ShenandoahHeap::marking_context() const {
   return _marking_context;
 }
 
+inline void ShenandoahHeap::set_cit(KlassInfoTable* cit) {
+  assert((_cit == nullptr && cit != nullptr) ||
+    (_cit != nullptr && cit == nullptr), "Initialize once & clear once");
```

Comment:
The assert message isn't accurate for your current usage. Suggest splitting into two asserts with a suitable message:
```
assert(_cit == nullptr || cit == nullptr, "Overwriting an existing histogram");
assert(_cit != nullptr || cit != nullptr, "Already cleared");
```